### PR TITLE
Limit cluster to 3 nodes

### LINKED ISSUE #2750: Cluster size should be limited to 3 servers
For the 0.9.0 release, clusters will be limited in size to 3 servers. We'll start the work to lift this limitation on the 0.9.1 release cycle. But in the meantime, we should forcibly limit the number of servers in a cluster.

If the user starts up a new server and attempts to join a cluster that already has 3 nodes, it should exit with a failure stating that the 0.9.0 release limits clusters to 3 servers.

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request restricts the maximum number of nodes in a cluster to 3.

Fixes #2750

### CODE REVIEW

File: meta/store.go [hunk 1]
```diff
@@ -129,6 +129,12 @@ func (s *Store) IDPath() string { return filepath.Join(s.path, "id") }
 
 // Open opens and initializes the raft store.
 func (s *Store) Open() error {
+	// Verify that no more than 3 peers.
+	// https://github.com/influxdb/influxdb/issues/2750
+	if len(s.peers) > 3 {
```
Comment (otoolep) [inline meta/store.go]: When is `s.peers` increased? When does it hit 4?

Comment (otoolep) [inline meta/store.go]: What I mean is that for this to work, `peers` must include the guy who is joining, correct?

File: meta/errors.go [hunk 2]
```diff
@@ -8,6 +8,9 @@ var (
 
 	// ErrStoreClosed is returned when closing an already closed store.
 	ErrStoreClosed = errors.New("raft store already closed")
+
+	// ErrTooManyPeers is returned when more than 3 peers are used.
+	ErrTooManyPeers = errors.New("too many peers; influxdb v0.9.0 is limited to 3 clusters in a node")
```
Comment (pauldix) [inline meta/errors.go]: 3 nodes in a cluster? :)

**Comment by pauldix** [comment]
Does this actually stop it from joining or does it just error the server out if they've joined more than 3 to the raft cluster?

File: meta/store.go (hunk 1 above)
Comment (benbjohnson) [inline meta/store.go]: Peers are hardcoded in the configuration file. All nodes need to have the same set of peers so there's not really a sense of "joining" anymore. If you start the first server up and you list 4 peers then it'll immediately fail.

File: meta/errors.go (hunk 2 above)
Comment (benbjohnson) [inline meta/errors.go]: Fixed:  0c0859f :)

**Comment by benbjohnson** [comment]
@pauldix You have to specify all peers in the config file so it will fail immediately even when the first node tries to start up.

**Comment by pauldix** [comment]
@benbjohnson ugh, that's really ugly. I guess this means a bunch of work needs to be done in 0.9.1 to add joining a cluster back in as a concept?

The config file thing won't work in the real world. Otherwise how are new nodes going to join a cluster?

**Comment by benbjohnson** [comment]
@pauldix I don't think it'll be a bunch of work. We still need the initial peer set so that nodes can wait for a quorum before creating the initial cluster.

**Comment by pauldix** [comment]
@benbjohnson cool, I just want to make sure that when we add that in, people running an 0.9.0 cluster will be able to upgrade without breaking config files or anything like that.

**Pull request merged**